### PR TITLE
Workflow updates

### PR DESCRIPTION
**Pull request by cmoylan**
Apply the requested changes from the following to our fork of the workflow gem.
* https://www.pivotaltracker.com/story/show/139951585
* https://github.com/avantcredit/avant-basic/pull/16805

### CODE REVIEW

File: test/new_versions/persistence_test.rb [hunk 1]
```diff
@@ -22,8 +22,6 @@ class PersistenceTestOrder < ActiveRecord::Base
     state :shipped
   end
 
-  attr_accessible :title # protecting all the other attributes
-
```
Comment (cmoylan) [inline test/new_versions/persistence_test.rb:26]: Deprecated in favor of strong params

File: test/advanced_hooks_and_validation_test.rb [hunk 2]
```diff
@@ -47,7 +47,10 @@ class Article < ActiveRecord::Base
         fields_to_validate = meta[:validates_presence_of]
         if fields_to_validate
           validations = Proc.new {
-            errors.add_on_blank(fields_to_validate) if fields_to_validate
+            fields_to_validate.each do |field|
+              value = read_attribute_for_validation(field)
+              errors.add(field, :empty) if value.nil? || value.empty?
+            end
```
Comment (cmoylan) [inline test/advanced_hooks_and_validation_test.rb:53]: `add_on_blank` is deprecated/removed

File: test/new_versions/persistence_test.rb [hunk 3]
```diff
@@ -50,12 +48,12 @@ def assert_state(title, expected_state, klass = PersistenceTestOrder)
...
   end
 
-  test 'ensure other dirty attributes are not saved on state change' do
+  test 'ensure other dirty attributes are saved on state change' do
     o = assert_state 'order6', 'accepted'
     o.title = 'going to change the title'
     assert o.changed?
     o.ship!
-    assert o.changed?, 'title should not be saved and the change still stay pending'
+    assert !o.changed?, 'title should be saved as a side effect of the state change'
```
Comment (ferdynton) [inline test/new_versions/persistence_test.rb]: Optionally, I think this could be `refute` instead of `assert !`

**Review by ferdynton — APPROVED** [review]
Sweet!

File: test/new_versions/persistence_test.rb (hunk 3 above)
Comment (cmoylan) [reply test/new_versions/persistence_test.rb]: Yes! `refute` is much cleaner.

**Pull request merged**